### PR TITLE
chore(scripts): improved local-to-CI linting

### PR DESCRIPTION
**Pull request by jzempel**
## Description

Add `.eslintignore` per [comment](https://github.com/zendeskgarden/react-components/pull/1312#discussion_r840859433)

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+!.storybook
+packages/**/dist
```
[the remark this answers is not in the document]
Comment (jzempel) [reply .eslintignore:2]: 😕 @Francois-Esquire this is the equivalent of adding `.storybook` to an allowlist – so it is being linted. Also, `node_modules` is a default that doesn't need to be added here. See https://eslint.org/docs/user-guide/configuring/ignoring-code#the-eslintignore-file for details.

**Review by hzhu — APPROVED** [review]

File: .eslintignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (hzhu) [reply .eslintignore:2]: > Lines preceded by ! are negated patterns that re-include a pattern that was ignored by an earlier pattern.

TIL. But what is the "earlier pattern" in this file? Seems like `!.storybook` is the first pattern and there is no earlier pattern?

[the remark this answers is not in the document]
Comment (jzempel) [reply .eslintignore:2]: > I'm not sure if this comment was correct...

From my testing, the implementation here is correct. Without `!.storybook` added to `.eslintignore`, the `.storybook` folder is ignored due to the ignore defaults:

- `node_modules/` is ignored.
- dot-files (except for `.eslintrc.*`), as well as dot-folders and their contents, are ignored.

Please see https://eslint.org/docs/user-guide/configuring/ignoring-code. I'd encourage you to pull the branch and play with the config, scripts, & lint-staged to observe the behaviors.

[the remark this answers is not in the document]
Comment (Francois-Esquire) [reply .eslintignore:2]: @hzhu I believe the earlier pattern is part of what is ignored by default (dot files/directories).

@jzempel You are correct, I misread the rule being removed.. thanks for breaking it down.

**Pull request merged**